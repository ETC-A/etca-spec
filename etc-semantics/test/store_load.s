;-mstrict
;8
.extension byte_operations

    movx %rx1,0
    mov  %rx0,0x0aff
    movx [%rx1],%rx0
    movx %rx2,[%rx1]
    movh %rh3,[%rh1]
    addx %rx1,1
    movh %rh4,[%rh1]  
    movh [%rh1],%rh0 ; note 1: not word-aligned
                     ; note 2: config checks *0 unchanged
end:
    jmp  end
