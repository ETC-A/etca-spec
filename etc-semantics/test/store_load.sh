set -u
./scripts/test.sh "$1" "$2" "$3" "test/store_load.bin" 16
