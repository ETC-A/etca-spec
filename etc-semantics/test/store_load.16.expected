<etc>
  <k>
    #halt "Halted successfully!" ~> .
  </k>
  <pc> 32788 </pc>
  <registers>
    0 |-> 2815
    1 |-> 1
    2 |-> 2815
    3 |-> -1
    4 |-> 10
    5 |-> 0
    6 |-> 0
    7 |-> 0
  </registers>
  <memory>
    0 |-> b"\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
    128 |-> b"Y Y\x02\\\x17\\\x1f\x1b\x04\x1aD\ndP!\n\x84\x0b\x04\x8e\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
  </memory>
  <flags>
    <carry>    false </carry>
    <zero>     false </zero>
    <negative> false </negative>
    <overflow> false </overflow>
  </flags>
  <machine-details>
    <cpuid> 16 </cpuid>
    <exten> 16 </exten>
    <reg-mode>  word </reg-mode>
    <reg-width> word </reg-width>
    <reg-count> 8 </reg-count>
    <evil-mode> false </evil-mode>
  </machine-details>
  <in> </in>
  <out> </out>
</etc>
